pkg/shortnames/shortnames.go

```diff
@@ -138,6 +138,8 @@ const (
 	rationaleUSR
 	// Resolved value has been selected by the user (via the prompt).
 	rationaleUserSelection
+	// Resolved value has been enforced to use Docker Hub (via SystemContext).
+	rationaleEnforcedDockerHub
```

Comment:
(I’ve been wondering if we even need this, or if we should just treat the input _exactly_ the same as fully-qualified.

Ideally (by hypothesis that there really are no short-name users, at least) users are _never_ going to see the short names and worry how they became `docker.io`-qualified.

OTOH it doesn’t hurt, sure, and if we _did_ ever encounter such users, the traceability is certainly nice.)